### ISSUE TITLE
Listbox: No accessible name (WCAG: 4.1.2)

### ISSUE THREAD

**Issue by lcameron-vs** [primefaces/primeng#11756]
### Describe the bug

Listbox unordered list has no accessible name. The `ul.p-listbox-list` element with `role="listbox"`, needs at least one of the following to comply with WCAG 4.1.2:

- `aria-label` attribute
- `aria-labelledby` attribute
- `title` attribute

https://www.w3.org/WAI/ARIA/apg/patterns/listbox/

### Environment

N/A

### Angular version

N/A

### PrimeNG version

14.0.0

### Build / Runtime

TypeScript

### Language

ALL

### Node version (for AoT issues node --version)

N/A

### Steps to reproduce the behavior

1. Go to http://primefaces.org/primeng/listbox
2. Right-click and inspect HTML on any listbox
3. Check attributes for `ul.p-listbox-list` element

Alternatively, analyse listbox with accessibility tools (e.g. axe DevTools).

### Expected behavior

Element with `role="listbox"` has an accessible name.

**Comment by cetincakiroglu** [comment]
Hi,

Accessibility is a major concern of the Prime UI libraries and PrimeNG is no exception. [PrimeTek](https://www.primetek.com.tr/) teams have initiated a significant process to review and enhance the accessibility features of the components. This guide documents the foundation of the general guidelines that PrimeNG will follow and each component documentation will have a separate Accessibility section that states the keyboard support, screen reader compatibility, the implementation details along with tips to achieve WCAG compliancy. This work has been initiated in Q2 2022 and planned to be completed by early Q3. PrimeNG will be the reference implementation which then will be ported to PrimeFaces, PrimeNG and PrimeVue.
